File: PRNumberBiz/app/src/main/java/com/pplus/prnumberbiz/apps/post/data/PostAdapter.java
//package com.pplus.prnumberbiz.apps.post.data;
//
//import android.content.Context;
//import android.content.Intent;
//import android.support.v4.view.ViewPager;
//import android.support.v7.widget.RecyclerView;
//import android.view.LayoutInflater;
//import android.view.View;
//import android.view.ViewGroup;
//import android.widget.ImageView;
//import android.widget.LinearLayout;
//import android.widget.TextView;
//
//import com.bumptech.glide.Glide;
//import com.bumptech.glide.request.RequestOptions;
//import com.pple.pplus.utils.part.utils.time.DateFormatUtils;
//import com.pplus.prnumberbiz.Const;
//import com.pplus.prnumberbiz.R;
//import com.pplus.prnumberbiz.apps.common.mgmt.LoginInfoManager;
//import com.pplus.prnumberbiz.apps.common.ui.custom.DirectionIndicator;
//import com.pplus.prnumberbiz.apps.common.ui.custom.GradeBar;
//import com.pplus.prnumberbiz.apps.post.ui.ReplyActivity;
//import com.pplus.prnumberbiz.core.code.common.EnumData;
//import com.pplus.prnumberbiz.core.network.model.dto.Page;
//import com.pplus.prnumberbiz.core.network.model.dto.Post;
//
//import java.text.SimpleDateFormat;
//import java.util.ArrayList;
//import java.util.Calendar;
//import java.util.Date;
//import java.util.HashMap;
//import java.util.List;
//
//
///**
// * Created by 김종경 on 2015-06-17.
// */
//public class PostAdapter extends RecyclerView.Adapter<PostAdapter.ViewHolder>{
//
//    private Context mContext;
//    private List<Post> mDataList;
//    private OnItemClickListener listener;
//    private static EnumData.PostType mPostType;
//    private int mTodayYear, mTodayMonth, mTodayDay;
//
//
//    public HashMap<Integer, Integer> viewPageStates = new HashMap<>();
//
//    public interface OnItemClickListener{
//
//        void onItemClick(int position);
//    }
//
//    public PostAdapter(Context context, EnumData.PostType postType){
//
//        setHasStableIds(true);
//
//        this.mContext = context;
//        this.mDataList = new ArrayList<>();
//        this.mPostType = postType;
//
//        Calendar c = Calendar.getInstance();
//        mTodayYear = c.get(Calendar.YEAR);
//        mTodayMonth = c.get(Calendar.MONTH);
//        mTodayDay = c.get(Calendar.DAY_OF_MONTH);
//    }
//
//    @Override
//    public long getItemId(int position){
//
//        return mDataList.get(position).getNo();
//    }
//
//    public void setOnItemClickListener(OnItemClickListener listener){
//
//        this.listener = listener;
//    }
//
//    public Post getItem(int position){
//
//        return mDataList.get(position);
//    }
//
//    public List<Post> getDataList(){
//
//        return mDataList;
//    }
//
//    public void add(Post data){
//
//        if(mDataList == null) {
//            mDataList = new ArrayList<>();
//        }
//        mDataList.add(data);
//        notifyDataSetChanged();
//    }
//
//    public void addAll(List<Post> dataList){
//
//        if(this.mDataList == null) {
//            this.mDataList = new ArrayList<>();
//        }
//
//        this.mDataList.addAll(dataList);
//        notifyDataSetChanged();
//    }
//
//    public void replaceData(int position, Post data){
//
//        mDataList.remove(position);
//        mDataList.add(position, data);
//        notifyDataSetChanged();
//    }
//
//    public void clear(){
//
//        mDataList = new ArrayList<>();
//        notifyDataSetChanged();
//    }
//
//    public void setDataList(List<Post> dataList){
//
//        this.mDataList = dataList;
//        notifyDataSetChanged();
//    }
//
//    static class ViewHolder extends RecyclerView.ViewHolder{
//
//        private ImageView image_post_profileImage;
//        ViewPager pager_image;
//        View layout_post_profile, layout_image, layout_post_grade, image_post_sns;
//        DirectionIndicator indicator_post;
//        TextView text_post_name, text_title, text_contents, text_regDate, text_comment, text_share;
//        GradeBar gradebar_review_post;
//
//        public ViewHolder(View itemView){
//
//            super(itemView);
//            layout_post_profile = itemView.findViewById(R.id.layout_post_profile);
//            image_post_profileImage = (ImageView) itemView.findViewById(R.id.image_post_profileImage);
//            text_post_name = (TextView) itemView.findViewById(R.id.text_post_name);
//            pager_image = (ViewPager) itemView.findViewById(R.id.pager_post_image);
//            layout_image = itemView.findViewById(R.id.layout_post_image);
//            indicator_post = (DirectionIndicator) itemView.findViewById(R.id.indicator_post);
//            text_title = (TextView) itemView.findViewById(R.id.text_post_title);
//            text_title.setSingleLine();
//            text_contents = (TextView) itemView.findViewById(R.id.text_post_contents);
//            text_regDate = (TextView) itemView.findViewById(R.id.text_post_regDate);
//            text_comment = (TextView) itemView.findViewById(R.id.text_post_comment);
//            text_share = (TextView) itemView.findViewById(R.id.text_post_share);
//            layout_post_grade = itemView.findViewById(R.id.layout_post_grade);
//            image_post_sns = itemView.findViewById(R.id.image_post_sns);
//            image_post_sns.setVisibility(View.GONE);
////            layout_product.setVisibility(View.GONE);
//            gradebar_review_post = itemView.findViewById(R.id.gradebar_review_post);
//            switch (mPostType) {
//                case pr:
//                case sns:
//                    layout_post_profile.setVisibility(View.GONE);
//                    text_share.setVisibility(View.VISIBLE);
//                    itemView.findViewById(R.id.view_post_share_bar).setVisibility(View.VISIBLE);
//                    break;
//                case review:
//                    layout_post_profile.setVisibility(View.VISIBLE);
//                    text_share.setVisibility(View.GONE);
//                    itemView.findViewById(R.id.view_post_share_bar).setVisibility(View.GONE);
//                    break;
//            }
//        }
//    }
//
//    @Override
//    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType){
//
//        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_post, parent, false);
//        return new ViewHolder(v);
//    }
//
//    @Override
//    public void onViewRecycled(final ViewHolder holder){
//
//        super.onViewRecycled(holder);
//        viewPageStates.put(holder.getAdapterPosition(), holder.pager_image.getCurrentItem());
//
//    }
//
//    @Override
//    public void onBindViewHolder(final ViewHolder holder, int position){
//
//        final Post item = mDataList.get(position);
//        holder.text_title.setText(item.getSubject());
//
//        holder.text_contents.setText("" + item.getContents());
//
//        if(item.getType().equals(EnumData.PostType.sns.name())){
//            holder.image_post_sns.setVisibility(View.VISIBLE);
//        }else {
//            holder.image_post_sns.setVisibility(View.GONE);
//        }
//
//        if(item.getAttachList() != null && item.getAttachList().size() > 0) {
//            holder.text_contents.setMaxLines(2);
//            holder.layout_image.setVisibility(View.VISIBLE);
//            holder.pager_image.setVisibility(View.VISIBLE);
//            PostImagePagerAdapter imageAdapter = new PostImagePagerAdapter(mContext, item.getAttachList());
//            holder.pager_image.setAdapter(imageAdapter);
//            holder.indicator_post.setVisibility(View.VISIBLE);
//            holder.indicator_post.removeAllViews();
//            holder.indicator_post.build(LinearLayout.HORIZONTAL, item.getAttachList().size());
//            holder.pager_image.addOnPageChangeListener(new ViewPager.OnPageChangeListener(){
//
//                @Override
//                public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels){
//
//                }
//
//                @Override
//                public void onPageSelected(int position){
//
//                    holder.indicator_post.setCurrentItem(position);
//                }
//
//                @Override
//                public void onPageScrollStateChanged(int state){
//
//                }
//            });
//
//            holder.pager_image.setCurrentItem(viewPageStates.containsKey(position) ? viewPageStates.get(position) : 0);
//            imageAdapter.setListener(new PostImagePagerAdapter.OnItemClickListener(){
//
//                @Override
//                public void onItemClick(int pos){
//
//                    if(listener != null) {
//                        listener.onItemClick(holder.getAdapterPosition());
//                    }
//                }
//            });
//        } else {
//            holder.text_contents.setMaxLines(5);
//            holder.indicator_post.removeAllViews();
//            holder.indicator_post.setVisibility(View.GONE);
//            holder.pager_image.setVisibility(View.GONE);
//            holder.pager_image.setAdapter(null);
//            holder.layout_image.setVisibility(View.GONE);
//        }
//
//        if(item.getAuthor().getProfileImage() != null){
//            Glide.with(mContext).load(item.getAuthor().getProfileImage().getUrl()).apply(new RequestOptions().centerCrop().placeholder(R.drawable.prnumber_default_img).error(R.drawable.prnumber_default_img)).into(holder.image_post_profileImage);
//        }
//
//        holder.text_post_name.setText("" + item.getAuthor().getNickname());
//
//
//        holder.text_comment.setText(mContext.getString(R.string.word_reply)+ " " + item.getCommentCount());
////        holder.text_post_price.setPaintFlags(holder.text_post_price.getPaintFlags() | Paint.STRIKE_THRU_TEXT_FLAG);
//
//        try {
//            Date d = DateFormatUtils.PPLUS_DATE_FORMAT.parse(item.getRegDate());
//            Calendar c = Calendar.getInstance();
//            c.setTime(d);
//
//            int year = c.get(Calendar.YEAR);
//            int month = c.get(Calendar.MONTH);
//            int day = c.get(Calendar.DAY_OF_MONTH);
//
//            if(mTodayYear == year && mTodayMonth == month && mTodayDay == day) {
//                SimpleDateFormat output = new SimpleDateFormat("a HH:mm");
//                holder.text_regDate.setText(output.format(d));
//            } else {
//                SimpleDateFormat output = new SimpleDateFormat("yyyy.MM.dd");
//                holder.text_regDate.setText(output.format(d));
//            }
//
//        } catch (Exception e) {
//
//        }
//
////        if(mPostType.equals(EnumData.PostType.review)){
////            if(item.getProperties() != null && StringUtils.isNotEmpty(item.getProperties().getStarPoint())){
////                holder.layout_post_grade.setVisibility(View.VISIBLE);
////                holder.gradebar_review_post.build(item.getProperties().getStarPoint());
////            }else{
////                holder.layout_post_grade.setVisibility(View.GONE);
////            }
////        }
//
//        holder.text_comment.setOnClickListener(new View.OnClickListener(){
//
//            @Override
//            public void onClick(View view){
//                Intent intent = new Intent(mContext, ReplyActivity.class);
//                intent.putExtra(Const.POST_NO, item.getNo());
//                mContext.startActivity(intent);
//            }
//        });
//
//        holder.text_share.setOnClickListener(new View.OnClickListener(){
//
//            @Override
//            public void onClick(View view){
//                share(item);
//            }
//        });
//
//        holder.itemView.setOnClickListener(new View.OnClickListener(){
//
//            @Override
//            public void onClick(View view){
//
//                if(listener != null) {
//                    listener.onItemClick(holder.getAdapterPosition());
//                }
//            }
//        });
//    }
//
//    private void share(Post post){
//
//        String subjectText = null;
//        Page page = LoginInfoManager.getInstance().getUser().getPage();
//        subjectText = page.getName();
//
//        String shareText = post.getSubject() + "\n" + mContext.getString(R.string.format_msg_invite_url, "news.php?boardNo=" + post.getNo());
//
//        Intent intent = new Intent(Intent.ACTION_SEND);
//        intent.setAction(Intent.ACTION_SEND);
//        intent.setType("text/plain");
//        //        intent.putExtra(Intent.EXTRA_SUBJECT, subjectText);
//        intent.putExtra(Intent.EXTRA_TEXT, shareText);
//
//        Intent chooserIntent = Intent.createChooser(intent, mContext.getString(R.string.word_share_post));
//        //        chooserIntent.putExtra(Intent.EXTRA_INITIAL_INTENTS, shareIntentList.toArray(new Parcelable[]{}));
//        mContext.startActivity(chooserIntent);
//    }
//
//    @Override
//    public int getItemCount(){
//
//        return mDataList.size();
//    }
//}

File: PRNumber/app/src/main/java/com/pplus/prnumberuser/apps/post/data/PostAdapter.kt
//package com.pplus.prnumberuser.apps.post.data
//
//import android.content.Context
//import android.content.Intent
//import androidx.viewpager.widget.ViewPager
//import androidx.recyclerview.widget.RecyclerView
//import android.view.LayoutInflater
//import android.view.View
//import android.view.ViewGroup
//import android.widget.LinearLayout
//import com.bumptech.glide.Glide
//import com.bumptech.glide.request.RequestOptions
//import com.pplus.utils.part.format.FormatUtil
//import com.pplus.utils.part.logs.LogUtil
//import com.pplus.utils.part.utils.StringUtils
//import com.pplus.utils.part.utils.time.DateFormatUtils
//import com.pplus.prnumberuser.Const
//import com.pplus.prnumberuser.R
//import com.pplus.prnumberuser.apps.common.builder.AlertBuilder
//import com.pplus.prnumberuser.apps.common.builder.OnAlertResultListener
//import com.pplus.prnumberuser.apps.friend.ui.SameFriendActivity
//import com.pplus.prnumberuser.apps.post.ui.MyPostActivity
//import com.pplus.prnumberuser.apps.post.ui.PostReplyActivity
//import com.pplus.prnumberuser.core.code.common.EnumData
//import com.pplus.prnumberuser.core.network.model.dto.Post
//import com.pplus.prnumberuser.core.util.PplusCommonUtil
//import kotlinx.android.synthetic.main.item_page_post.view.*
//import java.text.SimpleDateFormat
//import java.util.*
//
///**
// * Created by imac on 2018. 1. 8..
// */
//class PostAdapter : RecyclerView.Adapter<PostAdapter.ViewHolder> {
//
//    var mContext: Context? = null
//    var mDataList: MutableList<Post>? = null
//    var listener: OnItemClickListener? = null
//    var mTopVisible = true
//    internal var mTodayYear: Int = 0
//    internal var mTodayMonth: Int = 0
//    internal var mTodayDay: Int = 0
//
//    interface OnItemClickListener {
//
//        fun onItemClick(position: Int)
//    }
//
//    constructor(context: Context, topVisible:Boolean) : super() {
//        this.mContext = context
//        this.mDataList = ArrayList()
//        val c = Calendar.getInstance()
//        mTodayYear = c.get(Calendar.YEAR)
//        mTodayMonth = c.get(Calendar.MONTH)
//        mTodayDay = c.get(Calendar.DAY_OF_MONTH)
//        mTopVisible = topVisible
//    }
//
//    fun setOnItemClickListener(listener: OnItemClickListener) {
//
//        this.listener = listener
//    }
//
//    fun getItem(position: Int): Post {
//
//        return mDataList!![position]
//    }
//
//    fun getDataList(): MutableList<Post>? {
//
//        return mDataList
//    }
//
//    fun add(data: Post) {
//
//        if (mDataList == null) {
//            mDataList = ArrayList<Post>()
//        }
//        mDataList!!.add(data)
//        notifyDataSetChanged()
//    }
//
//    fun addAll(dataList: List<Post>) {
//
//        if (this.mDataList == null) {
//            this.mDataList = ArrayList<Post>()
//        }
//
//        this.mDataList!!.addAll(dataList)
//        notifyDataSetChanged()
//    }
//
//    fun replaceData(position: Int, data: Post) {
//
//        mDataList!!.removeAt(position)
//        mDataList!!.add(position, data)
//        notifyDataSetChanged()
//    }
//
//    fun clear() {
//
//        mDataList = ArrayList<Post>()
//        notifyDataSetChanged()
//    }
//
//    fun setDataList(dataList: MutableList<Post>) {
//
//        this.mDataList = dataList
//        notifyDataSetChanged()
//    }
//
//    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
//
//        val layout_profile = itemView.layout_post_profile
//        val image_profile = itemView.image_page_post_profile
//        val text_name = itemView.text_page_post_name
//        val layout_image = itemView.layout_page_post_image
//        val pager_image = itemView.pager_page_post_image
//        val indicator_post = itemView.indicator_page_post
//        val text_title = itemView.text_page_post_title
//        val text_contents = itemView.text_page_post_contents
//        val text_regDate = itemView.text_page_post_regDate
//        val text_comment = itemView.text_page_post_comment
//        val image_share = itemView.layout_page_post_share
//        val image_more = itemView.image_page_post_more
//        val layout_reply = itemView.layout_page_post_reply
//        val text_link = itemView.text_page_post_link
//        val layout_who = itemView.layout_page_post_who
//
//        init {
//            itemView.layout_page_post_bottom.visibility = View.VISIBLE
//            itemView.layout_page_post_detail_bottom.visibility = View.GONE
//            text_contents.maxLines = 2
//            if (itemView.context is MyPostActivity) {
//                layout_profile.visibility = View.GONE
//            }
//        }
//    }
//
//    override fun getItemCount(): Int {
//        return mDataList!!.size
//    }
//
//    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
//        val item = mDataList!![position]
//
//        if(mTopVisible){
//            holder.layout_profile.visibility = View.VISIBLE
//
//            if(item.author != null && item.author!!.page != null){
//                Glide.with(holder.itemView.context).load(item.author!!.page!!.thumbnail).apply(RequestOptions().centerCrop().placeholder(R.drawable.img_post_profile_default).error(R.drawable.img_post_profile_default)).into(holder.image_profile)
//
//                holder.image_profile.setOnClickListener {
//                    val location = IntArray(2)
//                    it.getLocationOnScreen(location)
//                    val x = location[0] + it.width / 2
//                    val y = location[1] + it.height / 2
//                    PplusCommonUtil.goPage(holder.itemView.context, item.author!!.page!!, x, y)
//                }
//
//                holder.text_name.text = item.author!!.page!!.name
//            }
//        }else{
//            holder.layout_profile.visibility = View.GONE
//        }
//
//
//        if (StringUtils.isNotEmpty(item.subject)) {
//            holder.text_title.visibility = View.VISIBLE
//            holder.text_title.text = item.subject
//        } else {
//            holder.text_title.visibility = View.GONE
//        }
//
//        holder.text_contents.text = item.contents
//        holder.text_comment.text = PplusCommonUtil.fromHtml(holder.itemView.context.getString(R.string.html_reply, FormatUtil.getMoneyType(item.commentCount.toString())))
//
//        try {
//            val d = DateFormatUtils.PPLUS_DATE_FORMAT.parse(item.regDate)
//            val c = Calendar.getInstance()
//            c.time = d
//
//            val year = c.get(Calendar.YEAR)
//            val month = c.get(Calendar.MONTH)
//            val day = c.get(Calendar.DAY_OF_MONTH)
//
//            if (mTodayYear == year && mTodayMonth == month && mTodayDay == day) {
//                val output = SimpleDateFormat("a HH:mm", Locale.getDefault())
//                holder.text_regDate.text = output.format(d)
//            } else {
//                val output = SimpleDateFormat("yyyy.MM.dd", Locale.getDefault())
//                holder.text_regDate.text = output.format(d)
//            }
//
//        } catch (e: Exception) {
//            holder.text_regDate.text = ""
//        }
//
//        holder.image_share.setOnClickListener {
//            share(item)
//        }
//
//        holder.layout_reply.setOnClickListener {
//            val intent = Intent(mContext, PostReplyActivity::class.java)
//            intent.putExtra(Const.DATA, item)
//            intent.flags = Intent.FLAG_ACTIVITY_SINGLE_TOP or Intent.FLAG_ACTIVITY_CLEAR_TOP
//            mContext?.startActivity(intent)
//        }
//
//        if (item.author?.haveSameFriends!!) {
//            holder.layout_who.visibility = View.GONE
//        } else {
//            holder.layout_who.visibility = View.GONE
//        }
//
//        holder.layout_who.setOnClickListener {
//            val intent = Intent(holder.itemView.context, SameFriendActivity::class.java)
//            val location = IntArray(2)
//            it.getLocationOnScreen(location)
//            val x = location[0] + it.width / 2
//            val y = location[1] + it.height / 2
//            intent.putExtra(Const.X, x)
//            intent.putExtra(Const.Y, y)
//            intent.putExtra(Const.DATA, item.author)
//            intent.flags = Intent.FLAG_ACTIVITY_SINGLE_TOP or Intent.FLAG_ACTIVITY_CLEAR_TOP
//            holder.itemView.context.startActivity(intent)
//        }
//
//        holder.image_more.setOnClickListener {
//            postAlert(item, position - 1)
//        }
//
//        if (item.attachList != null && item.attachList!!.isNotEmpty()) {
////            holder.text_text_title.maxLines = 5
//            holder.layout_image.visibility = View.VISIBLE
//            holder.pager_image.visibility = View.VISIBLE
//            val imageAdapter = PostImagePagerAdapter(holder.itemView.context, item.attachList!!)
//            holder.pager_image.adapter = imageAdapter
//
//            if (item.attachList!!.size > 1) {
//                holder.indicator_post.visibility = View.VISIBLE
//            } else {
//                holder.indicator_post.visibility = View.GONE
//            }
//
//            holder.indicator_post.removeAllViews()
//            holder.indicator_post.build(LinearLayout.HORIZONTAL, item.attachList!!.size)
//            holder.pager_image.addOnPageChangeListener(object : androidx.viewpager.widget.ViewPager.OnPageChangeListener {
//
//                override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {
//
//                }
//
//                override fun onPageSelected(position: Int) {
//
//                    holder.indicator_post.setCurrentItem(position)
//                }
//
//                override fun onPageScrollStateChanged(state: Int) {
//
//                }
//            })
//
////            holder.pager_image.currentItem = if (viewPageStates.containsKey(position)) viewPageStates.get(position) else 0
//            imageAdapter.setListener(object : PostImagePagerAdapter.OnItemClickListener {
//                override fun onItemClick(position: Int) {
//                    if (listener != null) {
//                        listener!!.onItemClick(holder.adapterPosition)
//                    }
//                }
//            })
////            imageAdapter.setListener {
////                if (listener != null) {
////                    listener!!.onItemClick(holder.adapterPosition)
////                }
////            }
//        } else {
////            holder.text_text_title.maxLines = 5
//            holder.indicator_post.removeAllViews()
//            holder.indicator_post.visibility = View.GONE
//            holder.pager_image.visibility = View.GONE
//            holder.pager_image.adapter = null
//            holder.layout_image.visibility = View.GONE
//        }
//
//        if(StringUtils.isNotEmpty(item.articleUrl)){
//            holder.text_link.visibility = View.VISIBLE
//        }else{
//            holder.text_link.visibility = View.GONE
//        }
//
//        holder.text_link.setOnClickListener {
//            if(StringUtils.isNotEmpty(item.articleUrl)){
//                PplusCommonUtil.openChromeWebView(holder.itemView.context, item.articleUrl!!)
//            }
//        }
//
//        holder.itemView.setOnClickListener {
//            if (listener != null) {
//                listener!!.onItemClick(holder.adapterPosition)
//            }
//        }
//    }
//
//    private fun postAlert(post: Post, position: Int) {
//
//        val builder = AlertBuilder.Builder()
//        builder.setLeftText(mContext?.getString(R.string.word_cancel))
//
//        builder.setContents(mContext?.getString(R.string.word_share), mContext?.getString(R.string.msg_report))
//
//        builder.setStyle_alert(AlertBuilder.STYLE_ALERT.LIST_BOTTOM).setOnAlertResultListener(object : OnAlertResultListener {
//
//            override fun onCancel() {
//
//            }
//
//            override fun onResult(event_alert: AlertBuilder.EVENT_ALERT) {
//
//                when (event_alert.getValue()) {
//                    1 -> share(post)
//                    2 -> {
//                        PplusCommonUtil.report(mContext!!, EnumData.REPORT_TYPE.article, post.no)
//                    }
//                }
//            }
//        }).builder().show(mContext)
//    }
//
//    fun share(item: Post) {
//        var contents = item.contents!!
//
//        if (contents.length > 50) {
//            contents = contents.substring(0, 50)
//        }
//        val shareText = contents + "\n" + mContext?.getString(R.string.format_msg_page_url, "news.php?postNo=" + item.no!!)
//
//        val intent = Intent(Intent.ACTION_SEND)
//        intent.action = Intent.ACTION_SEND
//        intent.type = "text/plain"
//        intent.putExtra(Intent.EXTRA_TEXT, shareText)
//
//        val chooserIntent = Intent.createChooser(intent, mContext?.getString(R.string.word_share_post))
//        mContext?.startActivity(chooserIntent)
//    }
//
//    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
//        val v = LayoutInflater.from(parent.context).inflate(R.layout.item_page_post, parent, false)
//        return ViewHolder(v)
//    }
//}